test/tests.spec.ts

```diff
@@ -2979,19 +2865,15 @@ describe("passport-saml /", function () {
         // Fake the current date to be after the valid time range
         fakeClock.restore();
         fakeClock = sinon.useFakeTimers(Date.parse("2014-05-28T00:19:08Z"));
-
-        samlObj.validatePostResponse(container, function (err, profile, logout) {
-          try {
-            should.exist(err);
-            err!.message!.should.match("SAML assertion expired");
-            done();
-          } catch (err2) {
-            done(err2);
-          }
-        });
+        try {
+          await samlObj.validatePostResponseAsync(container);
+          false.should.eql(true, "Should not succeed");
```

Comment:
Is there a reason we have to use this pattern instead of https://shouldjs.github.io/#should-throws?

Reply:
it does not work for async functions. Should.js is a deprecated module

Reply:
You can use the pattern I highlighted above instead:

```js
import assert = require("assert");
assert.rejects(samlObj.validatePostResponseAsync(container), /Invalid signature/);
```